datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -2076,4 +2098,25 @@ mod tests {
 
         Ok(())
     }
+    #[test]
+    fn test_change_redundant_column() -> Result<()> {
+        let t1_field_1 = DFField::new_unqualified("a", DataType::Int32, false);
+        let t2_field_1 = DFField::new_unqualified("a", DataType::Int32, false);
+        let t1_field_2 = DFField::new_unqualified("b", DataType::Int32, false);
+        let t2_field_2 = DFField::new_unqualified("b", DataType::Int32, false);
+
```

Comment:
Would it be possible to add one more field named "a" here (to show a counter other than `:1`)?